File: TerminBA/TerminBA.Services/Recommender/RecommendationResult.cs
namespace TerminBA.Services.Recommender
{
    public class RecommendationResult
    {
        public int FacilityId { get; set; }

        public string FacilityName { get; set; } = string.Empty;

        public string SportCenterName { get; set; } = string.Empty;

        public DateTime StartTime { get; set; }

        public DateTime EndTime { get; set; }

        public decimal Price { get; set; }

        public float Score { get; set; }

        public List<string> Reasons { get; set; } = new();

        public bool IsPersonalized { get; set; } = true;
    }
}

File: TerminBA/TerminBA.UI/terminba_mobile/lib/model/recommendation_result.dart
import 'package:json_annotation/json_annotation.dart';

part 'recommendation_result.g.dart';

@JsonSerializable()
class RecommendationResult {
  final int facilityId;
  final String facilityName;
  final String sportCenterName;
  final DateTime startTime;
  final DateTime endTime;
  final double price;
  final double score;
  final List<String> reasons;
  final bool isPersonalized;

  const RecommendationResult({
    required this.facilityId,
    required this.facilityName,
    required this.sportCenterName,
    required this.startTime,
    required this.endTime,
    required this.price,
    required this.score,
    required this.reasons,
    required this.isPersonalized,
  });

  factory RecommendationResult.fromJson(Map<String, dynamic> json) =>
      _$RecommendationResultFromJson(json);

  Map<String, dynamic> toJson() => _$RecommendationResultToJson(this);
}
